src/main/java/com/google/step/servlets/ChannelPromoCodeServlet.java

```diff
@@ -13,19 +13,19 @@
 import java.util.List;
 import java.util.Optional;
 
-@WebServlet("/promo-code")
-public class PromoCodeServlet extends HttpServlet {
+@WebServlet("/channel/promo-code")
```

Comment:
nit: maybe /channel/promo-code**s** would be a more appropriate URI since we are returning multiple promo-codes